libtenzir/src/execution_node.cpp

```diff
@@ -621,16 +641,23 @@ struct exec_node_state {
     // 1. The operator's generator is not yet completed.
     // 2. The operator has one of the three following reasons to do work:
     //   a. The upstream operator has completed.
-    //   b. The operator has issued demand that is not yet answered.
-    //   c. The operator can produce output independently from receiving input.
-    //   d. The operator has input it can consume.
-    const auto should_continue = instance->it != instance->gen.end()  // (1)
-                                 and (not previous                    // (2a)
-                                      or issue_demand_inflight        // (2b)
-                                      or op->input_independent()      // (2c)
-                                      or not inbound_buffer.empty()); // (2d)
+    //   b. The operator has upstream demand and can produce output
```

Comment:
You mean "downstream" right?

Reply:
Yes, absolutely. There is no such thing as upstream demand.

```suggestion
    //   b. The operator has downstream demand and can produce output
```